### DISCUSSION TITLE
Use after/ftplugin/lua.lua to reset shiftWidth to 2

### DISCUSSION THREAD

**Discussion by jeffschwartz** [LazyVim/LazyVim#4960] [Q&A]
I have added the file .config/nvim/after/ftplugin/lua.lua to my LazyVim setup and in this file I have the following code to change tab width from 4 to 2 for .lua files:

```lua
print("here!")
local set = vim.opt_local
set.shiftwidth = 2
set.tabstop = 2
```

However, after restarting neovim the tab width in .lua files is still 4. I know this because if I save a .lua file that has 2 character indenting those indents are changed to a spacing of 4 when saved via autoformat. What appears to be happening is that the global setting is applied after my ftplugin lua.lua file is called.

Am I doing something wrong? Perhaps my expectation that using an ftplugin file is the correct way to implement this?

Thanks in advance for any assistance you can provide.

**Comment by dpetka2001** [comment]
The default is already set to 2. Did you change it in your `options.lua` file? Also launch Neovim with `nvim -V1` and then do `:verbose set shiftwidth?` to show you where it's last set from.

**Reply by dpetka2001** [reply]
Also the indentation that happens on save autoformat depends on the formatter that you use. So, also check your `stylua.toml` settings, except if you use some other Lua formatter, so best check its docs.

**Reply by jeffschwartz** [reply]
Thank you for your very quick response and yes, it was my stylua.toml file. I don't recall modifying it but someone obviously did and `indent_width` was set to 4. I apologize for wasting your time and I am so grateful for your contribution to this project. I am going to close this as resolved.
